frame/staking/src/lib.rs

```diff
@@ -2791,10 +2791,6 @@ where
 		slash_fraction: &[Perbill],
 		slash_session: SessionIndex,
 	) -> Result<Weight, ()> {
-		if !Self::can_report() {
-			return Err(());
-		}
-
```

Comment:
To actually achieve the goal of `An offence should be immediately processed by the staking pallet`, this method **OnOffenceHandler::on_offence** must return the **Weight** instead **Result<Weight, ()>**, and it is actually possible now, but change this return type will be a big breaking change.

Reply:
You can change the function signature back to just return the weight.

Reply:
That's what I wanted to hear, thanks! 🎉